File: server/Entities/OrderAggregate/Order.cs
namespace server.Entities.OrderAggregate;

public class Order : BaseEntity
{
    public int? TableId { get; set; }
    public Table? Table { get; set; }
    public required string TableCode { get; set; }
    public string OrderType { get; set; } = "Dine In";
    public int? NumberOfPeople { get; set; }
    public string BuyerName { get; set; } = string.Empty;
    public string BuyerEmail { get; set; } = string.Empty;
    public string? BuyerNote { get; set; } = string.Empty;
    public DateTime OrderDate { get; set; } = DateTime.UtcNow;
    public List<OrderItem> OrderItems { get; set; } = [];
    public decimal Subtotal { get; set; }
    public decimal Discount { get; set; }
    public OrderStatus OrderStatus { get; set; }
    public decimal GetTotal() => Subtotal - Discount;
}

File: client/Guest-Ordering-UI/src/models/order.ts
export interface Order {
    id: number
    tableId?: number
    tableName?: string
    tableCode: string
    buyerName: string
    buyerEmail: string
    buyerNote: string
    orderType?: string
    numberOfPeople?: number
    orderDate: string
    items: OrderItem[]
    subtotal: number
    discount: number
    orderTotal: number
    orderStatus: string
}

export interface OrderItem {
    id: number
    menuItemId: number
    name: string
    imageUrl: string
    price: number
    quantity: number
    note: string
}

export type CreateOrderItem = {
    menuItemId: number
    quantity: number
    note?: string
}

export type CreateOrderRequest = {
    tableCode: string
    orderType: "Dine In"
    numberOfPeople?: number
    buyerName?: string
    buyerEmail?: string
    buyerNote: string
    discount: number
    items: CreateOrderItem[]
}

export type ServiceResult<T> = {
    success: boolean
    message: string
    data: T
}